### PR TITLE
ci: Workflow for web and registry publishing

### PR DESCRIPTION
**Pull request by hcnp**
This replaces the old pipeline at: https://jenkins-test.admin.ehealth.sundhed.dk/job/common-implementationguide/job/master/

Currently the web page, that is published by the pipeline when merging to master, can be reached at https://fut-infrastructure.github.io/fut-ig-website/

I've tried to describe how to make a release on the [README page](https://github.com/fut-infrastructure/implementation-guide/pull/17/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

I'm unsure whether to keep https://docs.ehealth.sundhed.dk/ on a AWS s3, and then just link to the new IG web - maybe with a new url like https://ig.ehealth.sundhed.dk/ - or if we should move everything to GitHub pages.

### CODE REVIEW

**Review by tyfoni-systematic — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -66,7 +66,16 @@ The command for doing the publication is the following:
 Execution takes a couple of minutes. Once done, the content of the `$(pwd)/fut-ig-website` must be committed and pushed to the origin. Now your publication is done.
 
 ## Publication trigger rules - WORK IN PROGRESS
-Publications needs to be triggered by human interaction. Either by approval of merging to master or by tagging on master.
+
+When a new publication version is to be created, the following rules apply:
+
+- All changes must be created as a pull request on master
+- They can only be merged when they have been reviewed/approved
+- When all changes have been merged to master, a tag with the new SemVer version
```
[the remark this answers is not in the document]
Comment (hcnp) [reply README.md:74]: Yes. That is done here: https://github.com/fut-infrastructure/implementation-guide/pull/17/files#diff-e81e13daadd1745de51d8b8d85fce1fcd9be355732b64d60a6b56e18c28388caR53

I don't know about the change log. It is not published until master is updated and a tag is created. Maybe @jkiddo can answer.

Release notes are meant to be made in https://github.com/fut-infrastructure/implementation-guide/blob/master/release-notes.md

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  ig_cmd: java -Xmx4g -jar /input-cache/publisher.jar
+
+jobs:
+  # Check if the IG transpiles without errors
+  check-transpile:
+    name: Transpile IG
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/fut-infrastructure/ig-publisher/ig-publisher:latest
```
Comment (kdhtrifork) [inline .github/workflows/publish.yaml:22]: `latest` is a bit dangerous - can we pin it?

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        run: $ig_cmd -ig .
+
+  # Transpile and publish the IG, if it was pushed to main with a tag
+  release:
+    name: Transpile and publish to the IG release
+    # A release is defined as a tag pushed to main
+    if: ${{ github.event_name == 'push' && github.ref_type == 'tag' }}
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/fut-infrastructure/ig-publisher/ig-publisher:latest
```
Comment (kdhtrifork) [inline .github/workflows/publish.yaml:38]: `latest` is a bit dangerous - can we pin it?

Comment (hcnp) [reply .github/workflows/publish.yaml:38]: We could, but it's kind of pinned by creating new versions with tags in https://github.com/fut-infrastructure/ig-publisher/blob/main/.github/workflows/container-build.yaml

We always want to use the newest publisher version. We can always temporarily pin it to an old one if something breaks.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkiddo) [reply README.md:74]: @tyfoni-systematic I'd argue that 'unreleased' should stay in the release notes of the branch that constitutes the upcoming/next release. As pointed out, the release notes are most likely subject to be changed a bit. There are essentially two approaches. A format looking like e.g. this: https://fhir.ch/ig/ch-core/changelog.html which is sort of the same as we have today, or we transition to a format like e.g. https://hl7.dk/fhir/core/history.html - meaning that the release notes only contain information about the current release and not all history - as it is a relase note concerning _this_ release.

File: .github/workflows/publish.yaml (hunk 3 above)
Comment (jkiddo) [reply .github/workflows/publish.yaml:38]: I agree with @kdhtrifork but it seems contained for now.

**Review by jkiddo — APPROVED** [review]

**Pull request merged**